packages/e2e-tests/specs/experiments/blocks/navigation.test.js

```diff
@@ -506,4 +507,53 @@ describe( 'Navigation', () => {
 		// Expect a Navigation Block with a link for "A really long page name that will not exist".
 		expect( await getEditedPostContent() ).toMatchSnapshot();
 	} );
+
+	it( 'loads frontend code only if the block is present', async () => {
+		// Mock the response from the Pages endpoint. This is done so that the pages returned are always
+		// consistent and to test the feature more rigorously than the single default sample page.
+		await mockPagesResponse( [
+			{
+				title: 'Home',
+				slug: 'home',
+			},
+			{
+				title: 'About',
+				slug: 'about',
+			},
+			{
+				title: 'Contact Us',
+				slug: 'contact',
+			},
+		] );
+
+		const isScriptLoaded = await page.evaluate(
+			() =>
+				null !== document.querySelector( 'script[src*="frontend.js"]' )
```

Comment:
It might give false negatives when there are multiple frontend scripts loaded. All scripts from Gutenberg should have their handle included in the `<script` tag and you can use the direct match.